test/e2e/setup/setup.go

```diff
@@ -164,6 +166,45 @@ func deployOperator() error {
 	); err != nil {
 		return errors.Errorf("error building operator deployment: %s", err)
 	}
+
+	// TODO: check with timeout for whether the deployment.Status.ReadyReplicas field is 0
+	// use kubernetes client to get deployment object
+	// read status
+	// return error if 0
+	// TODO: check if we could get Name from service account
+	// TODO: check if actualReplicas is getting updated inside the loop
+	dep := appsv1.Deployment{}
+
+	// retrieve required replicas -- they will already be there
+	if err := e2eutil.TestClient.Get(context.TODO(),
+		types.NamespacedName{Name: "mongodb-kubernetes-operator",
+			Namespace: e2eutil.OperatorNamespace},
+		&dep); err != nil {
+		return errors.Errorf("error building operator deployment: %s", err)
+	}
+	requiredReplicas := *dep.Spec.Replicas
+
+	// poll to check whether actual number of replicas matches required number of replicas
+	deadline := time.Now().Add(10 * time.Second)
+	for {
+		if err := e2eutil.TestClient.Get(context.TODO(),
+			types.NamespacedName{Name: "mongodb-kubernetes-operator",
+				Namespace: e2eutil.OperatorNamespace},
+			&dep); err != nil {
+			return errors.Errorf("error building operator deployment: %s", err)
+		}
+		actualReplicas := dep.Status.ReadyReplicas
+		if actualReplicas == requiredReplicas || time.Now().After(deadline) {
+			break
+		}
+	}
+
+	// check if now the condition is fulfilled
+	actualReplicas := dep.Status.ReadyReplicas
+	if actualReplicas != requiredReplicas {
+		return errors.New("error building operator deployment: actual number of replicas does not match required number of replicas")
+	}
+
```

Comment:
I was just looking at if we can use some existing function from k8s and I found this: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait . I think we can use this and pass the function of getting the deployment and checking it's status. I think it will make the code more robust and easy to follow.